tests/test_read.py

```diff
@@ -96,3 +97,46 @@ def test_read_pillow_frombytes(heif_file):
 def test_read_10_bit(path):
     heif_file = pyheif.read(path)
     image = create_pillow_image(heif_file)
+
+
+@pytest.mark.parametrize("path", heif_files + hif_files)
+def test_open_and_load(path):
+    heif_file = pyheif.open(path)
+    assert heif_file.size[0] > 0
+    assert heif_file.size[1] > 0
+    assert heif_file.has_alpha is not None
+    assert heif_file.mode is not None
+    assert heif_file.bit_depth is not None
+
+    assert heif_file.data is None
+    assert heif_file.stride is None
+
+    if path.name == 'arrow.heic':
+        assert heif_file.metadata
+        assert heif_file.color_profile
+
+    res = heif_file.load()
+    assert heif_file is res
+    assert heif_file.data is not None
+    assert heif_file.stride is not None
+    assert len(heif_file.data) >= heif_file.stride * heif_file.size[1]
+    assert type(heif_file.data[:100]) == bytes
+
+    # Subsequent calls don't change anything
+    res = heif_file.load()
+    assert heif_file is res
+    assert heif_file.data is not None
+    assert heif_file.stride is not None
+
+
+@pytest.mark.parametrize("path", heif_files + hif_files)
+def test_open_and_load_data_collected(path):
+    data = path.read_bytes()
+    heif_file = pyheif.open(data)
+    
+    # heif_file.load() should work even if there is no other refs 
+    # to the source data.
+    data = None
+    gc.collect()
+
```

Comment:
More curious than anything, what's the need to manually collect here?

Reply:
The images are loading by `_libheif_cffi.lib.heif_context_read_from_memory_without_copy` function. This means, that the memory data we provide as a source should stay alive until any decoding processes will be finished.

This test checks this. We release our reference to the data (`data = None`) and additionally call garbage collector for the case if there were circular references to data or we are working in the PyPy, where no reference counting exist.

The last line of the test: we call `heif_file.load()` to check it works under such conditions.